py/desimeter/fieldmodel.py

```diff
@@ -92,17 +93,53 @@ def read_guide_stars_catalog(self,filename,max_sep_arcsec = 2.) :
         # here we could do some conversion of column names
         catalog = Table.read(filename)
 
+        if "mjd_obs" in catalog.dtype.names :
+            self.mjd = np.mean(catalog["mjd_obs"])
+            log.info("use mjd={} from catalog['mjd_obs']".format(self.mjd))
+
         if ( not  "xcentroid" in catalog.dtype.names ) or ( not "ra_gaia" in catalog.dtype.names ) :
             log.error("I can only deal with Aaron's catalogs with columns xcentroid,ycentroid,ra_gaia,dec_gaia, sorry")
             raise RuntimeError("I can only deal with Aaron's catalogs with columns xcentroid,ycentroid,ra_gaia,dec_gaia, sorry")
 
-        max_sep_arcsec = 2.
         log.info("selection stars for which we have a good match (< {} arcsec)".format(max_sep_arcsec))
 
-        dra  = (catalog["ra"]-catalog["ra_gaia"])*cosd(catalog["dec_gaia"]) * 3600. # arcsec
-        ddec = (catalog["dec"]-catalog["dec_gaia"]) * 3600. # arcsec
-        dr = np.hypot(dra, ddec)
-        selection = (dr<2) # arcsec
+        if all([_ in catalog.columns for _ in ['pmra', 'pmdec', 'ref_epoch']]):
+
+            if self.mjd is None :
+                log.error("Cannot compute proper motion correction because mjd=None")
+                raise RuntimeError("Cannot compute proper motion correction because mjd=None")
+
+            # if proper motions and reference epochs are there
+            pmra = catalog['pmra']
+            pmdec = catalog['pmdec']
+            # if unknown, zero out the pms
+            pmra[~np.isfinite(pmra)] = 0
+            pmdec[~np.isfinite(pmdec)] = 0
+
+            cur_year = Time(self.mjd, format='mjd').to_value(format='jyear')
+            # observation time in decimal years (like 2020.3)
+            ref_epoch = catalog['ref_epoch']
+            dra  = (cur_year - ref_epoch) * pmra / 3600e3 / cosd(catalog['dec_gaia'])
+            ddec = (cur_year - ref_epoch) * pmdec / 3600e3
+
+            # add pm and rename columns
+            catalog['ra_gaia'] += dra
+            catalog['dec_gaia'] += ddec
+            catalog.rename_column('ra_gaia','ra_gaia_with_pm')
+            catalog.rename_column('dec_gaia','dec_gaia_with_pm')
+
+            match_dra  = (catalog["ra"]-catalog['ra_gaia_with_pm'])*cosd(catalog['dec_gaia_with_pm']) * 3600. # arcsec
```

Comment:
Minor gripe. Personally I'd remove code duplication here and do something like
```
ra_column = 'ra_gaia_with_pm'
dec_column = 'dec_gaia_with_pm'
```
and then do 
```
match_dra = catalog['ra']-catalog[ra_column] )*cosd(...)
```
shared for two codepaths.